analyzer/consensus/consensus.go

```diff
@@ -588,6 +595,95 @@ func (m *Main) queueNodeEvents(batch *storage.QueryBatch, data *storage.Registry
 	return nil
 }
 
+func (m *Main) queueRegistryEvents(batch *storage.QueryBatch, data *storage.RegistryData) error {
```

Comment:
With all the new `queue*Events()` methods, can we delete `queueEventInserts()` and `extractEventData()` in this file? IIUC their functionality was refactored into these new methods.

Reply:
Oh I missed all of these blocks:
```
if hash != nil {
			continue
		}
```

So the new methods only process events that are not associated with a tx? Then `queueEventInserts()` should only process events that _are_ associated with a tx, I suppose, though I don't see why we wouldn't just process both kinds of events here.

Reply:
> Then queueEventInserts() should only process events that are associated with a tx

Yes. This is the case before and after this PR.

> why we wouldn't just process both kinds of events here.

Unfortunately, we are doing two passes. First, we obtain the `tx_index` information based on the position of the associated `tx` in the array from `GetTransactionsWithResults`. Next, we get the events without a `tx` from the individual service backends.

I believe this is the improved algorithm mentioned [here](https://github.com/oasisprotocol/oasis-indexer/pull/188#issuecomment-1307863567).

Hmm, perhaps I can make this clearer by extrapolating the switch case statements into helpers.

Reply:
Danke. Revised as suggested.